src/verifiable-credential.js

```diff
@@ -39,24 +40,26 @@ class VerifiableCredential {
   }
 
   /**
-   * Add a context to this Credential's context array
+   * Add a context to this Credential's context array. Duplicates are omitted.
    * @param {string|object} context - Context to add to the credential context array
    * @returns {VerifiableCredential}
    */
   addContext(context) {
     ensureObjectWithKeyOrURI(context, '@context', 'context');
-    this.context.push(context);
+    this.context = getUniqueElementsFromArray([...this.context, context], JSON.stringify);
```

Comment:
This will iterate over the collection `this.context` each time a context is added and stringify each element to detect duplicates. An efficient alternative would be to keep the collection as a Set and convert back to array and destringify in `toJSON`.

Reply:
Thanks! 
I believe your approach is more efficient if `addContext` is called more often than `toJSON`. If the opposite applies, the doing the conversions in `addContext` should be more efficient. 
It's hard to foresee which scenario will take place so we have to guess:
my guess is that `toJSON` will be called more often than `addContext` because it is possible to go through the entire issuing process without needing to add a new context, whereas `toJSON` is already called internally to sign and to verify even with no user interaction. Hence keeping my proposed approach would execute the conversions less often and be more efficient.

Reply:
I see that the context `https://www.w3.org/2018/credentials/examples/v1` is almost always needed which is added using the call and a more appropriate type will usually be added. But its fine, we are not adding 10s of contexts or types so let it be.